tools/apigee-sackmesser/cmd/export/export.sh

```diff
@@ -58,46 +58,111 @@ sackmesser list "organizations/$organization/apis" | jq -r -c '.[]|.' | while re
         loginfo "download developer app: $appId for developer: $email"
-        mkdir -p "$export_folder/developerApps/$email"
-        sackmesser list "organizations/$organization/developers/$email/apps/$(urlencode "$appId")" > "$export_folder"/developerApps/"$email"/"$appId".json
+        full_app=$(sackmesser list "organizations/$organization/developers/$email/apps/$(urlencode "$appId")")
+        echo "$full_app" | jq 'del(.credentials)' > "$export_folder/temp/developerApps/$email/$appId".json
+        echo "$full_app" | jq -r -c '.credentials[]' | while read -r credential; do
+            appkey=$(echo "$credential" | jq -r '.consumerKey')
+            echo "$credential" | jq  --arg APP_NAME "$appId" '. + { name: $APP_NAME } | . + { apiProducts: [.apiProducts[]?.apiproduct] }' > "$export_folder/temp/importKeys/$email/$appkey.json"
+        done
     done
+
+    if ls "$export_folder/temp/developerApps/$email"/*.json 1> /dev/null 2>&1; then
+        jq -n "{ \"$email\": [inputs] }" "$export_folder/temp/developerApps/$email"/*.json > "$export_folder/temp/apps/$email.json"
+    else
+        echo "{ \"$email\": [] }" > "$export_folder/temp/apps/$email.json"
+        loginfo "No Apps for Developer: $email"
+    fi
+
+    if ls "$export_folder/temp/importKeys/$email"/*.json 1> /dev/null 2>&1; then
+        jq -n "{ \"$email\": [inputs] }" "$export_folder/temp/importKeys/$email"/*.json > "$export_folder/temp/importKeys/$email.json"
+    else
+        echo "{ \"$email\": [] }" > "$export_folder/temp/importKeys/$email.json"
+        loginfo "No App Keys for Developer: $email"
+    fi
 done
 
+jq -n '[inputs]' "$export_folder/temp/developers"/*.json > "$export_folder/config/resources/edge/org/developers.json"
+jq -n '[inputs] | add' "$export_folder/temp/apps"/*.json > "$export_folder/config/resources/edge/org/developerApps.json"
+jq -n '[inputs] | add' "$export_folder/temp/importKeys"/*.json > "$export_folder/config/resources/edge/org/importKeys.json"
+
+mkdir -p "$export_folder/temp/apiproducts"
 sackmesser list "organizations/$organization/apiproducts" | jq -r -c '.[]|.' | while read -r product; do
     loginfo "download API product: $product"
-    mkdir -p "$export_folder/apiproducts"
-    sackmesser list "organizations/$organization/apiproducts/$(urlencode "$product")" > "$export_folder"/apiproducts/"$product".json
+    sackmesser list "organizations/$organization/apiproducts/$(urlencode "$product")" > "$export_folder/temp/apiproducts/$product".json
 done
+jq -n '[inputs]' "$export_folder/temp/apiproducts"/*.json > "$export_folder/config/resources/edge/org/apiProducts.json"
 
-sackmesser list "organizations/$organization/keyvaluemaps" > "$export_folder"/kvms.json
+mkdir -p "$export_folder/temp/keyvaluemaps"
+sackmesser list "organizations/$organization/keyvaluemaps" | jq -r -c '.[]|.' | while read -r kvmname; do
+    sackmesser list "organizations/$organization/keyvaluemaps/$kvmname" > "$export_folder/temp/keyvaluemaps/$kvmname".json
```

Comment:
I think it's worth checking if any of the KVMs have 1000 entries, as that's the limit for what the Edge API will return. In that case, it may be a good idea to `logwarn` the user that some entries in `$kvmname` may be missing from the export.

Context: find "1000" on https://apidocs.apigee.com/docs/key-value-maps/1/overview

Reply:
Nice one. Added a warning log as suggested